### PR TITLE
add fxcop analyzer

### PR DESCRIPTION
**Pull request by MS-TimothyMothra**
Add the FxCop Analyzer to Non-Test projects.

This analyzer identifies **SEVERAL** compiler errors. 
If you're bored one afternoon, feel free to pull this branch and fix a couple! :)

dreamwork

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
@MS-TimothyMothra this approach never works. I'd suggest merge fixes as you go. So you will not end up with enormous merge conflict in a few weeks.

**Comment by MS-TimothyMothra** [comment]
@SergeyKanzhelev I'm open to suggestions for a better approach. 
I thought this might be the easiest for the whole team to contribute to.

Across all our repos there are over 700 errors. It will almost certainly be required to resolve these before our next SDL review. We can discuss more offline.

**Comment by SergeyKanzhelev** [comment]
I'm suggesting to send fixes in separate PRs and merge them as you making them. Like you did 3 fixes - send them as a separate PR. Perhaps this may be a good instruction:

```
git checkout -b fxcop/feelBraveToday
git cherry-pick 9801d05246043bf99bccd7d336573601541460e3

... 
do the fixes
...

git revert <cherry-pick commit ID>
git push
```

Maybe fixing everything in one PR will also work as this repo has low traffic overall.

Thanks again for starting it! 👍

**Comment by MS-TimothyMothra** [comment]
@SergeyKanzhelev this is ready to be reviewed.
When @Dmitry-Matveev gets back, I want to discuss if we should wait to merge this with a Beta1 release.

**Comment by MS-TimothyMothra** [comment]
@Dmitry-Matveev this is a medium PR. :)

File: Common.props [hunk 1]
```diff
@@ -21,7 +21,8 @@
     <Optimize>true</Optimize>
     <RunCodeAnalysis>true</RunCodeAnalysis>
     <CodeAnalysisTreatWarningsAsErrors>true</CodeAnalysisTreatWarningsAsErrors>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <!-- Need to disable WarningsAsErrors until all FxCop issues are fixed. -->
+    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (SergeyKanzhelev) [inline Common.props]: do you have work item for this so we wouldn't forget?

File: src/NLogTarget/NLogTarget.csproj [hunk 2]
```diff
@@ -29,6 +29,10 @@
     <PackageReference Include="Desktop.Analyzers" Version="1.1.0">
       <PrivateAssets>All</PrivateAssets>
     </PackageReference>
+    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="2.6.1">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
```
Comment (SergeyKanzhelev) [inline src/NLogTarget/NLogTarget.csproj]: this is default as I understand. Why you need it explicitly?

Comment (MS-TimothyMothra) [reply src/NLogTarget/NLogTarget.csproj]: This was inserted by default. I haven't tested without.

Comment (SergeyKanzhelev) [reply src/NLogTarget/NLogTarget.csproj]: I'd suggest to remove to avoid confusion

**Pull request merged**